core/schema/query.go

```diff
@@ -40,8 +40,7 @@ func (s *querySchema) Install() {
 		// JSON and written to a core.File. This is currently used internally for calling
 		// module SDKs and is thus hidden the same way the rest of introspection is hidden
 		// (via the magic __ prefix).
-		dagql.NodeFunc("__schemaJSONFile", s.schemaJSONFile).
-			Impure("Return value depends on what the caller has installed in the server.").
+		dagql.NodeFuncWithCacheKey("__schemaJSONFile", s.schemaJSONFile, dagql.CachePerCall).
```

Comment:
As an idea for how we could cache these in the future ... everytime we modify client state, we could increase a client counter, and then mixin that counter + the client id to the target.

Maybe you already have ideas for how to do this though!

Reply:
Yeah that's not a bad idea. I was figuring when a schema is installed/served we could calculate a digest based on which modules are installed and use that as a content-based digest for this, which *technically* is slightly better in that we could re-use the same cache for this file when two clients end up with the exact same schema, but practically speaking either approach works and is mostly the same.